pulp_rpm/app/models/package.py

```diff
@@ -351,4 +351,10 @@ def list_to_createrepo_c(l):
         package.url = getattr(self, PULP_PACKAGE_ATTRS.URL)
         package.version = getattr(self, PULP_PACKAGE_ATTRS.VERSION)
 
+        if checksum_type:
+            pkgId = getattr(self.contentartifact_set.first().artifact, checksum_type.lower(), None)
+            if pkgId:
+                package.checksum_type = checksum_type.lower()
+                package.pkgId = pkgId
```

Comment:
@goosemania @ipanova I didn't know how to handle the case when we set `package_checksum_type` and the sync is `on_demand`